src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,61 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validUser(user);
+        validLogin(user);
+        validPassword(user);
+        validAge(user);
+        return storageDao.add(user);
+    }
+
+    public void validUser(User user) {
+        if (user == null) {
+            throw new ValidationException("User can't be equaled null");
+        }
+    }
+
+    public void validLogin(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("The data base contains user with such login. "
+                    + "Please change your login.");
+        }
```

Comment:
place this check at the end of this method, what's the point of checking null or invalid credential? and you can get NPE